syntax = "proto3";
package execute;




import "model/tool/callback.proto";
import "google/protobuf/struct.proto";

/*
ExecuteFlow请求
*/
message ExecuteFlowRequest {
message Metadata {
    /*
    环境
    */
    string type = 1 ;
    /*
    描述
    */
    string desc = 2 ;
}
        
    
    /*
    流程执行时输入参数
  "flowInputs": {
      "@agents": [
        {
          "ip": "192.168.100.163",
          "instanceId": "5caee6c67d86a"
        }
      ],
      "test1": "test params",
      "ddddd": "5caee6c67d86a",
      "test2": "develop_1"
    },

    */
    google.protobuf.Struct flowInputs = 1 ;
    /*
    自动填充参数
    */
    google.protobuf.Struct instanceMap = 2 ;
    /*
    是否发送notify通知
    */
    string needNotify = 3 ;
    /*
    父流程任务ID
    */
    string parentTaskId = 4 ;
    /*
    流程执行成功后的回调,有中间态回调,(部署在使用)
    */
    tool.Callback callback = 5 ;
    /*
    流程全部执行完回调,有别于callback字段
    */
    tool.Callback endCallback = 6 ;
    /*
    流程ID
    */
    string flowId = 7 ;
    /*
    版本号(时间戳-秒)
    */
    int32 version = 8 ;
    /*
    流程名称
    */
    string name = 9 ;
    /*
    流程环境变量
    */
    google.protobuf.Struct flowEnv = 10 ;
    /*
    流程环境类型
    */
    Metadata metadata = 11 ;
    /*
    流程标签
    */
    repeated string tags = 12 ;
}

/*
ExecuteFlow返回
*/
message ExecuteFlowResponse {
    /*
    流程任务ID
    */
    string taskId = 1 ;
}

/*
ExecuteFlowApi返回
*/
message ExecuteFlowResponseWrapper {
    /*
    返回码
    */
    int32 code = 1 ;
    /*
    返回码解释
    */
    string codeExplain = 2 ;
    /*
    错误详情
    */
    string error = 3 ;
    /*
    返回数据
    */
    execute.ExecuteFlowResponse data = 4 ;
}

